scripts/generateForDay.sh

```diff
@@ -1,41 +1,168 @@
+	echo -e "      's' is to ${MSG1} a startrail"
+	echo -e "      't' is to ${MSG1} a timelapse"
+	echo -e "    If you don't specify k, s, or t, all three will be ${MSG2}."
+	echo -e "${NC}"
+	exit 0
 fi
 
-DATE=$1
-DIR="$ALLSKY_HOME/images/$DATE"
+if [ $# -gt 1 ] ; then
+	DO_K="false"
+	DO_S="false"
+	DO_T="false"
+	while [ $# -gt 1 ]
+	do
+		if [ "${1}" = "-k" ] ; then
+			DO_K="true"
+		elif [ "${1}" = "-s" ] ; then
+			DO_S="true"
+		elif [ "${1}" = "-t" ] ; then
+			DO_T="true"
+		else
+			echo "Unknown image type: '${1}'; ignoring."
+		fi
+		shift
+	done
+else
+	DO_K="true"
+	DO_S="true"
+	DO_T="true"
+fi
+
+DATE="${1}"
+DATE_DIR="${ALLSKY_IMAGES}/${DATE}"
+if [ ! -d "${DATE_DIR}" ] ; then
+	echo -e "${ME}: ${RED}ERROR: '${DATE_DIR}' not found!${NC}"
+	exit 2
+fi
 
-# Generate timelapse from collected images
-echo -e "${ME}: Generating Keogram\n"
-mkdir -p ${DIR}/keogram/
-../keogram ${DIR}/ $EXTENSION ${DIR}/keogram/keogram-$DATE.jpg
-echo -e "\n"
+if [ "${DO_K}" = "true" ] ; then
+	KEOGRAM_FILE="keogram-${DATE}.${EXTENSION}"
+	UPLOAD_FILE="${DATE_DIR}/keogram/${KEOGRAM_FILE}"
+	if [ "${TYPE}" = "GENERATE" ]; then
+		# Don't include ${ME} since this script is always executed manually
+		echo -e "===== Generating Keogram"
+		mkdir -p "${DATE_DIR}/keogram"
 
-# Generate startrails from collected images. Treshold set to 0.1 by default in config.sh to avoid stacking over-exposed images
-echo -e "${ME}: Generating Startrails\n"
-mkdir -p ${DIR}/startrails
-../startrails ${DIR}/ $EXTENSION $BRIGHTNESS_THRESHOLD ${DIR}/startrails/startrails-$DATE.jpg
-echo -e "\n"
+		# The keogram command outputs one line for each of the many hundreds of files,
+		# and this adds needless clutter to the output, so send output to a tmp file so we
+		# can output the number of images.
+		TMP="${ALLSKY_TMP}/keogramTMP.txt"
+		"${ALLSKY_HOME}/keogram" -d "${DATE_DIR}" -e ${EXTENSION} -o "${UPLOAD_FILE}" ${KEOGRAM_PARAMETERS} > "${TMP}"
+		[ $? -eq 0 ] && echo -e "Completed"
+	else
+		if [ -s "${UPLOAD_FILE}" ]; then
+			# If the user specified a different name for the destination file, use it.
+			if [ "${KEOGRAM_DESTINATION_NAME}" != "" ]; then
+				KEOGRAM_FILE="${KEOGRAM_DESTINATION_NAME}"
+			fi
+			echo "Uploading ${KEOGRAM_FILE}"
+			# KG == KeoGram.   "--silent" is for silent mode
+			"${ALLSKY_SCRIPTS}/upload.sh" --silent "${UPLOAD_FILE}" "${KEOGRAM_DIR}" "${KEOGRAM_FILE}" "KG"
+			[ $? -eq 0 ] && echo "${KEOGRAM_FILE} uploaded"
+		else
+			echo -en "${YELLOW}"
+			echo -n "ERROR: Keogram file '${UPLOAD_FILE}' not found; skipping"
+			echo -e "${NC}"
+		fi
+	fi
+	echo -e "\n"
+fi
+
+if [ "${DO_S}" = "true" ] ; then
+	STARTRAILS_FILE="startrails-${DATE}.${EXTENSION}"
+	UPLOAD_FILE="${DATE_DIR}/startrails/${STARTRAILS_FILE}"
+	if [ "${TYPE}" = "GENERATE" ]; then
+		echo -e "===== Generating Startrails, threshold=${BRIGHTNESS_THRESHOLD}"
+		mkdir -p "${DATE_DIR}/startrails"
+
+		# The staretrails command outputs one line for each of the many hundreds of files,
```

Comment:
same here. startrails runs mostly quietly except for the brightness statistics at the end, unless you use the `-v` flag.